app/controllers/mixins/actions/vm_actions/evacuate.rb

```diff
@@ -40,6 +40,7 @@ def evacuate_vm
 
           @sb[:action] = nil
           if @sb[:explorer]
+            @sb[:explorer] = nil
```

Comment:
won't this will break in explorer screens?

Reply:
`@sb[:explorer]` was `nil` even when we were in explorer screen. It may be weird but this is what I found when I was debugging the code, `nil` is the original value, before we decide to evacuate selected instances. And after the action, I am just moving it back to `nil`. We need this change for fixing redirecting issue mentioned in the description of this PR. Try to evacuate instances on explorer screen, then do the same with instances displayed in a nested list - after the action, you will hit the issue, without this change in the code.

Reply:
Sorry, nope, setting `@sb[:explorer] = nil` in all those actions just sounds dangerous.

Is there a good reason for it? What will happen if you don't clear it?

Reply:
Yes, it sounds dangerous but after long debugging and investigating... When just displaying Instances in explorer screen, `@sb[:explorer]` IS already `nil`, even when `@explorer` is true. `@sb[:explorer]` becomes true [right here](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/mixins/actions/vm_actions/evacuate.rb#L8), when evacuating instances. So why not to put it back after the action? Another reason is that if we don't do it, bug will occur, I've already mentioned it in the description of this PR. Shortly, try to evacuate instances displayed in explorer, then display instances in a nested list (non-explorer based screen) and evacuate (or migrate or reconfigure). After the action the screen will not be rendered properly because of wrong redirecting to the previous screen - because `@sb[:explorer]` was true. Also note that [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/6759/files#diff-d1dcfc1e06f051662384d5742d42014cR42) we cannot simply use `if @explorer`. It will always be nil.

Reply:
You're right, thanks! :)

`@sb[:explorer]` gets set by the action on the first request, to persist the explorer state,
and unset on the last request, after submit/cancel now.

The following call to `replace_right_cell` sets `@explorer = true` anyway, but this way the `@sb[:explorer]` logic gets cleared when leaving the form properly.

:+1:

Reply:
Right. And I was thinking.. maybe I should better move this cleanup to the separate issue/PR? WDYT? :)

Reply:
Up to you :) If these are 2 unrelated thigns, go for it :)